taskstats/src/nlas.rs

```diff
@@ -0,0 +1,289 @@
+        use TaskStatsCmdAttrs::*;
+        match self {
+            Pid(v) => ne::write_u32(buffer, *v),
+            TGid(v) => ne::write_u32(buffer, *v),
+            RegisterCPUMask(s) => {
+                buffer[..s.len()].copy_from_slice(s.as_bytes());
+                buffer[s.len()] = 0;
+            }
+            DeRegisterCPUMask(s) => {
+                buffer[..s.len()].copy_from_slice(s.as_bytes());
+                buffer[s.len()] = 0;
+            }
+        }
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for TaskStatsCmdAttrs {
+    fn parse(buf: &NlaBuffer<&'a T>) -> Result<Self, DecodeError> {
+        let payload = buf.value();
+        Ok(match buf.kind() {
+            TASKSTATS_CMD_ATTR_REGISTER_CPUMASK => Self::RegisterCPUMask(
+                parse_string(payload).context("invalid TASKSTATS_CMD_ATTR_REGISTER_CPUMASK value")?,
+            ),
+            TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK => Self::DeRegisterCPUMask(
+                parse_string(payload).context("invalid TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK value")?,
+            ),
+            TASKSTATS_CMD_ATTR_PID => {
+                Self::Pid(parse_u32(payload).context("invalid TASKSTATS_CMD_ATTR_PID value")?)
+            }
+            TASKSTATS_CMD_ATTR_TGID => {
+                Self::TGid(parse_u32(payload).context("invalid TASKSTATS_CMD_ATTR_TGID value")?)
+            }
+            kind => return Err(DecodeError::from(format!("Unknown NLA type: {}", kind))),
+        })
+    }
+}
+
+/*-------------------- Taskstats Events --------------------*/
+
+
+/// Event code definition 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum TaskStatsEventAttrs {
+    /// process id
+	Pid(i32),
+    /// Thread group id
+	TGid(i32),
+    /// taskstats structure
+    Stats(struct_tasksstats),
+    /// contains pid + stats
+    AggrPid(Statistics),
+    /// contains tgid + stats
+    AggrTGid(Statistics),
+    /// contains nothing
+    Null
+}
+
+const TASKSTATS_TYPE_PID:u16 = 1;		/* Process id */
+const TASKSTATS_TYPE_TGID:u16 = 2;		/* Thread group id */
+const TASKSTATS_TYPE_STATS:u16 = 3;		/* taskstats structure */
+const TASKSTATS_TYPE_AGGR_PID:u16 = 4;	/* contains pid + stats */
+const TASKSTATS_TYPE_AGGR_TGID:u16 = 5;	/* contains tgid + stats */
+const TASKSTATS_TYPE_NULL:u16 = 6;		/* contains nothing */
+
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+/// Taststats + additional information
+pub struct Statistics
+{
+    pub tgid:           i32,
+    pub pid:            i32,
+    pub timestamp_ns:   u64,
+    pub data:           struct_tasksstats
+}
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+///the actual struct containing statistics
+/* 
+    created by bindgen
+    use c reperesentation rather than NativeEndian (which is usually used in this crate) 
+    for correct field alignments 
+*/
+
+pub struct struct_tasksstats
```

Comment:
This is valid naming for rust struct

Reply:
Maybe, but it's the exact copy of the struct in the kernel. The name of the struct emphasizes this ... what do you suggest?